### ISSUE TITLE
`alt`/`else` block shown in parallel using teoz leads to wrong formatting

### ISSUE THREAD

**Issue by mxmlnkn** [plantuml/plantuml#591]
The alt/else block can't be rendered in parallel by itself, see also #269 but I accidentally got it somewhat rendered in parallel by having the alt/else block in another group. But, then, the header and separation line for the `else` block move to the top instead of being in the middle where it should be:

```plantuml
@startuml
!pragma teoz true

!theme crt-amber
skinparam backgroundColor #000000

participant "Random Name" as foo

hide footbox

foo -> foo : test

& opt message received
    alt REQUEST
        bossrpcp -> bossrpcp : request
    else RESPONSE
        bossrpcp -> bossrpcp : respond
    end
end

@enduml
```

Rendering for the source as shown:

![teoz-alt-else-bug-parallel](https://user-images.githubusercontent.com/6842824/125301519-5ab76780-e32b-11eb-96da-edec62bc0c15.png)

Rendering without the `&`:

![teoz-alt-else-bug-not-parallel](https://user-images.githubusercontent.com/6842824/125301405-407d8980-e32b-11eb-965a-e05ae62edc90.png)

**Comment by jimnelson372** [comment]
Hello @mxmlnkn and @The-Lum, I believe I have found the cause of this issues and I have implemented a solution.   Here is what my fixed version looks like.  (This is the exact script above, but commenting out the theme and backgroundColor instructions, as I only have the default theme in my test environment):

![testteozaltparallel](https://github.com/plantuml/plantuml/assets/66284321/be2d463f-8084-431a-8e72-4f763c9255c8)

I've tested that this fix works regardless of the order of the two parallel elements; in other words, `foo -> foo: test` can be moved to after the opt group and it still works.   My fix will work when an alt/else block is contained within an Opt group.  But there is an additional problem shown in issue #269 when trying to use an alt/else block directly as a parallel element.  I'm looking at that now.

By the way, you can see in the diagram I shared that the upper line of the self message is a little higher than the top of the Opt group.   There is another issue (#1791) where I noticed the  top lines of self messages were not matching up with other messages.   I will see if that can be adjusted, but it may need to wait on later fixes.

Regards,

Jim Nelson

**Comment by jimnelson372** [comment]
Hello @mxmlnkn and @The-Lum, I was able to find a solution for #269 as well.  I wrote a comment on that issue demonstrating it.  Let me know if anything still looks out of place with it.  Thanks.

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed on [V1.2024.6](https://github.com/plantuml/plantuml/releases/tag/v1.2024.6).

- Thanks @mxmlnkn for the report.
- Thanks @jimnelson372  for your work.

Regards.
